server/src/controllers/ingredientsCtrlr.ts

```diff
@@ -0,0 +1,226 @@
+import { type Request, type Response } from 'express';
+import axios from 'axios';
+import { load } from 'cheerio';
+
+const UNITS = new Set([
+  'teaspoon',
+  'tablespoon',
+  'cup',
+  'ounce',
+  'pound',
+  'gram',
+  'kilogram',
+  'milliliter',
+  'liter',
+  'clove',
+  'stalk',
+  'can',
+]);
+
+interface Ingredient {
+  name: string;
+  quantity: number;
+  unit: string;
+}
+
+// Given a certified unit string, turn it into lower and without an 's' at the end if it has one.
+function retUnit(word: string): string {
+  const lower = word.toLowerCase();
+
+  if (lower.endsWith('s')) {
+    return lower.slice(0, -1);
+  }
+
+  return lower;
+}
+
+// Make sure that the given string is a unit from the UNITS dictionary.

```

Comment:
Does this account for units like "tsp" which would be the equivalient of "teaspoon"

Reply:
No, but from looking at Allrecipe's website I haven't found an instance where they use the shortened versions.

Reply:
might be a good idea to include it just in case if it's not too much work, but ill leave that to ur discretion cuz i haven't looked at the allrecipes site much

Reply:
Aight I can add it. It doesn't take too long.